components/script/dom/headers.rs

```diff
@@ -535,6 +535,7 @@ pub fn extract_mime_type(headers: &HyperHeaders) -> Option<Vec<u8>> {
                 // Step 6.4
                 if temp_essence != essence {
                     charset = temp_charset.map(|c| c.to_string());
+                    temp_essence.clone_into(&mut essence);
```

Comment:
Should this not replace the below?

Reply:
Oh, nice catch. This was an oversight on my part which should be fixed now.